lib/katello/engine.rb

```diff
@@ -4,11 +4,13 @@ class Engine < ::Rails::Engine
 
     initializer 'katello.selective_params_parser', :before => :build_middleware_stack do |app|
       require 'katello/params_parser_wrapper'
-      app.middleware.swap(
-        ActionDispatch::ParamsParser,
-        Katello::ParamsParserWrapper,
-        ->(env) { env['PATH_INFO'] =~ /consumers/ && env['PATH_INFO'] =~ /profile|packages/ }
-      )
+      if Rails::VERSION::MAJOR == 4
+        app.middleware.swap(
+          ActionDispatch::ParamsParser,
+          Katello::ParamsParserWrapper,
+          ->(env) { env['PATH_INFO'] =~ /consumers/ && env['PATH_INFO'] =~ /profile|packages/ }
+        )
+      end
```

Comment:
I didn't follow very well what the wrapper does, but on Rails 5 (https://github.com/rails/rails/commit/38d2bf5fd1f3e014f2397898d371c339baa627b1) there's no ParamsParser - Foreman removed it from it's middleware too